src/core/thread/mle_router.cpp

```diff
@@ -2686,7 +2693,21 @@ ThreadError MleRouter::SendChildUpdateRequest(Child *aChild)
     Ip6::Address destination;
     Message *message;
 
+    if ((aChild->mMode & ModeTlv::kModeRxOnWhenIdle) == 0)
+    {
+        // for sleepy child, not send again if there is already queued indirect child update request message
+        for (message = mNetif.GetMeshForwarder().GetSendQueue().GetHead(); message; message = message->GetNext())
+        {
+            if (message->GetChildMask(mRouterRestoringChildIndex) &&
```

Comment:
Suggestion: Let's use the child index from the argument `aChild` here (using `mNetif.GetMle().GetChildIndex(aChild)`) instead of using the `mRouterRestoringChildIndex`.